src/connectivity_check/connectivity_check.go

```diff
@@ -353,19 +354,23 @@ func (c *connectivity) checkHost(conCheck connectivityCmd, outCh chan *models.Co
 	dataCh := make(chan any)
 	go c.l3CheckConnectivity(addresses, dataCh, conCheck)
 	go c.l2CheckConnectivity(dataCh, conCheck)
+	l2C := make(map[string]models.L2Connectivity)
```

Comment:
I think some background for the choice of using a map here is warranted and not so obvious

```suggestion
        // We use a map to de-duplicate arping responses from the same mac. They are redundant
        //  because the reason we're reporting more than one line of arping in the first place is to allow
        // the service to also detect devices in the network that have IP conflict with cluster hosts
	l2C := make(map[string]models.L2Connectivity)
```